infra/dlv/dlv-setup.sh

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+# Copyright 2016-2018 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.!/bin/bash
+
+SSH="ssh -o StrictHostKeyChecking=no"
+SCP="scp -o StrictHostKeyChecking=no"
+
+function usage() {
+    echo "Usage: $0 -h vch-address -p password -s -a" >&2
+    exit 1
+}
+
+while getopts "h:p:sa" flag
+do
+    case $flag in
+
+        h)
+            # Optional
+            export DLV_TARGET_HOST="$OPTARG"
+            ;;
+
+        p)
+            export SSHPASS="$OPTARG"
+            ;;
+
+        s)
+            COPY_SSH_KEY=true
+            ;;
+        a)
+            SSH_AUTHORIZED=true
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+shift $((OPTIND-1))
+
+if [ -z "${DLV_TARGET_HOST}" ]; then
+    usage
+fi
+
+if [ -z "${SSHPASS}" -a -z "${SSH_AUTHORIZED}" ]; then
+    usage
+fi
+
+if [ -n "${SSHPASS}" -a -z "${SSH_AUTHORIZED}" ]; then
+    SSH="sshpass -e ${SSH}"
+    SCP="sshpass -e ${SCP}"
+
+    if [ ! -f /usr/bin/sshpass ]; then
+        echo sshpass must be installed. Run \"apt-get install sshpass\"
+        exit 1
+    fi
+fi
+
+DLV_BIN="$GOPATH/bin/dlv"
+
+if [ -z "${GOPATH}" -o -z "${GOROOT}" ]; then
+    echo GOROOT and GOPATH should be set to point to the current GOLANG enironment
+    exit 1
+fi
+
+# copy dlv binary
+echo -n copying dlv binary..
```

Comment:
Is there a means of checking if delve is already set up? We don't want to repeatedly copy the same binaries over, particularly if it's in WDC or somewhere with similar bandwidth constraints.